spec/lib/auth0/mixins/httpproxy_spec.rb

```diff
@@ -148,6 +169,26 @@
           .and_raise(@exception)
         expect { @instance.send(http_method, '/test') }.to raise_error(Auth0::Unauthorized)
       end
+      
+      it "should raise Auth0::RateLimitEncountered on send http #{http_method} method
+        to path defined through HTTP when 429 status received" do
+        headers = {
+          'X-RateLimit-Limit'     => 10,
+          'X-RateLimit-Remaining' => 0,
+          'X-RateLimit-Reset'     => 1560564149
+        }
+        @exception.response = StubResponse.new({}, false, 429,headers)
+        allow(RestClient::Request).to receive(:execute).with(method: http_method,
+                                                             url: '/test',
+                                                             timeout: nil,
+                                                             headers: nil,
+                                                             payload: '{}')
+          .and_raise(@exception)
+        expect { @instance.send(http_method, '/test') }.to raise_error { |error|
+          expect(error).to be_a(Auth0::RateLimitEncountered)
+          expect(error).to have_attributes(reset: Time.at(1560564149))
```

Comment:
Maybe check that we also have the headers and the code? Either here or in another test.

Reply:
This is what I was referring to when I said I wasn't sure where to put tests that were general to `Auth0::Exception` or `Auth0::HTTPError`; the `code` and `headers` features are not specific to `Auth0::RateLimitEncountered` (to say nothing of the `error_data` feature), so I wasn't sure it made sense to test them in that context; that said, I can certainly add tests for those features there if desired.

Reply:
These tests have been added now.